staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -887,3 +887,79 @@ type MemorySwapConfiguration struct {
 	// +optional
 	SwapBehavior string `json:"swapBehavior,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CredentialProviderConfig is the configuration containing information about
```

Comment:
@liggitt WDYT? Is it awkward that these are in kubelet/config/v1beta1 now that folks want to move them to GA? Is it possible to create kubelet/config/v1 with only support for the CredentialProviderConfig (I can't remember if that'll work or yell about being unable to round trip the non-GA stuff through available API versions)?

Reply:
@ndixita was wondering the best path forward here

Reply:
> Is it awkward that these are in kubelet/config/v1beta1 now that folks want to move them to GA?

GA feature should correspond to GA config, yes

> Is it possible to create kubelet/config/v1 with only support for the CredentialProviderConfig

Yes, sparse versions work fine